tools/clang/include/clang/SPIRV/Type.h

```diff
@@ -0,0 +1,118 @@
+//===-- Type.h - SPIR-V Type --*- C++-*------------------------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+#ifndef LLVM_CLANG_SPIRV_TYPE_H
+#define LLVM_CLANG_SPIRV_TYPE_H
+
+#include <set>
+#include <unordered_set>
```

Comment:
we actually need \<set\> because we use an _ordered set_ of Decorations in a Type.
Inserting into an ordered set is less efficient than inserting into an unordered_set. (log(n) vs constant). But, having ordered sets makes comparison of Type objects much more efficient.
Note that insertion occurs only _once_ when the type is created, however, comparison could occur many many times.